src/os.rs

```diff
@@ -36,6 +36,30 @@ use rand_core::{RngCore, Error, ErrorKind, impls};
 /// Max OS X, and modern Linux) this may block very early in the init
 /// process, if the CSPRNG has not been seeded yet.[1]
 ///
+/// *Note*: many Unix systems provide `/dev/random` as well as `/dev/urandom`.
+/// This module uses `getrandom` if available, otherwise `/dev/urandom`, for
+/// the following reasons:
+///
+/// -   On Linux, `/dev/random` may block if entropy pool is empty;
+///     `/dev/urandom` will not block.  This does not mean that `/dev/random`
+///     provides better output than `/dev/urandom`; the kernel internally runs a
+///     cryptographically secure pseudorandom number generator (CSPRNG) based on
+///     entropy pool for random number generation, so the "quality" of
+///     `/dev/random` is not better than `/dev/urandom` in most cases.  However,
+///     this means that `/dev/urandom` can yield somewhat predictable randomness
+///     if the entropy pool is very small, such as immediately after first
+///     booting.  Linux 3.17 added the `getrandom(2)` system call which solves
```

Comment:
Maybe we should say a little less here, I am not sure this description is entirely up-to-date. Maybe we should just say that that the quality difference between `/dev/random` and `/dev/urandom` is a myth or at least outdated information. Making this a very short paragraph, as an addition to the list of system interfaces used.

Reply:
It's also about `getrandom`; I did consider just referring to the man page. It's just a cut-and-paste.